src/components/CategorizationRules/CategorizationRulesTable/CategorizationRulesTable.tsx

```diff
@@ -1,5 +1,158 @@
+import { BaseConfirmationModal } from '../../BaseConfirmationModal/BaseConfirmationModal'
+import { useArchiveCategorizationRule } from '../../../hooks/useCategorizationRules/useArchiveCategorizationRule'
+import { useLayerContext } from '../../../contexts/LayerContext/LayerContext'
+
+enum CategorizationRuleColumns {
+  Category = 'Category',
+  Counterparty = 'Counterparty',
+  Delete = 'Delete',
+}
+const COMPONENT_NAME = 'CategorizationRulesTable'
+
 export const CategorizationRulesTable = () => {
+  const [selectedRule, setSelectedRule] = useState<CategorizationRule | null>(null)
+  const [showDeletionConfirmationModal, setShowDeletionConfirmationModal] = useState(false)
+  const { trigger: archiveCategorizationRuleTrigger } = useArchiveCategorizationRule()
+  const { addToast } = useLayerContext()
+
+  const archiveCategorizationRule = useCallback(() => {
+    if (selectedRule?.id) {
+      archiveCategorizationRuleTrigger(selectedRule?.id).then(() => {
+        setShowDeletionConfirmationModal(false)
+      }).catch(() => {
+        addToast({ content: 'Failed to archive categorization rule', type: 'error' })
+      })
+    }
+  }, [addToast, archiveCategorizationRuleTrigger, selectedRule?.id])
+
+  const { data, isLoading, isError, size, setSize, refetch } = useListCategorizationRules({})
+  const categorizationRules = useMemo(() => data?.flatMap(({ data }) => data), [data])
+
+  const paginationMeta = data && data.length > 0 ? data[data.length - 1].meta.pagination : undefined
+  const hasMore = paginationMeta?.hasMore
+
+  const { currentCategorizationRulesPage: currentPage, setCurrentCategorizationRulesPage: setCurrentPage } = useSetCurrentCategorizationRulesPage()
+
+  const fetchMore = useCallback(() => {
+    if (hasMore) {
+      void setSize(size + 1)
+    }
+  }, [hasMore, setSize, size])
+
+  const paginationProps = useMemo(() => {
+    return {
+      initialPage: currentPage,
+      onSetPage: setCurrentPage,
+      pageSize: 10,
+      hasMore,
+      fetchMore,
+    }
+  }, [hasMore, fetchMore, currentPage, setCurrentPage])
+
+  const CategorizationRulesTableEmptyState = useCallback(() => {
+    return (
+      <DataState
+        status={DataStateStatus.allDone}
+        title='No rules found'
+        description='No categorization rules have been created yet. You will receive suggestions for rules to create as you categorize transactions in the bank feed.'
+        icon={<PencilRuler />}
+        spacing
+      />
+    )
+  }, [])
+
+  const CategorizationRulesTableErrorState = useCallback(() => (
+    <DataState
+      status={DataStateStatus.failed}
+      title='We couldn’t load your categorization rules'
+      description='An error occurred while loading your categorization rules. Please check your connection and try again.'
+      onRefresh={() => { void refetch() }}
+      spacing
+    />
+  ), [refetch])
+
+  const columnConfig: ColumnConfig<CategorizationRule, CategorizationRuleColumns> = ({
```

Comment:
There are anonymous functions crated on this config (onValueChange, onPress). I recommend wrapping this config in a `useMemo`.

So far right now I can only see one implementation (delete), so this will likely be better in a `useMemo`. If some portions of this config was actually intended to be static and doesn't require any references to component callbacks, then this can be defined outside of the component rendering code instead.